.github/actions/update-viablestrict/action.yml

```diff
@@ -85,8 +78,14 @@ runs:
       run: |
         set -ex
 
-        output=$(python ${GITHUB_WORKSPACE}/test-infra/.github/scripts/fetch_latest_green_commit.py --requires "${{ inputs.requires }}")
+        TEST_INFRA_PATH="${GITHUB_ACTION_PATH}/../../.."
```

Comment:
I have an uneasy feeling about this.  My understanding is that the runner keeps all the GHA it needs locally, and because we don't have ephemeral runner, `test-infra` is there because some other jobs have already fetched it.  The problem here is that I'm not sure if the version of test-infra is correct, and we could end up with a stale GHA

Reply:
You could test this out on an ephemeral linux runner and see if test-infra is there I think

Reply:
The run I linked as the test uses `runs-on: ubuntu-20.04`, I assume gha runners are as ephemeral as our linux runners?  Unless maybe gha runners are more special

The version did run my the code in my branch (it errored when the argument flags were wrong)